plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -233,6 +233,12 @@ func (plugin *InterfaceConfigurator) ConfigureVPPInterface(iface *intf.Interface
 		}
 	}
 
+	// set rx mode
+	err = vppcalls.SetRxMode(ifIdx, iface.RxMode, plugin.Log, plugin.vppCh, nil  )
```

Comment:
From the (very brief) VPP documentation:

This command is used to assign the RX packet processing mode (polling, interrupt, adaptive) of the a given interface, and optionally a given queue. If the 'queue' is not provided, the 'mode' is applied to all queues of the interface. **Not all interfaces support all modes**. To display the current rx-mode use the command 'show interface rx-placement'.

In proto file, rx mode is not bound to any specific interface type. So you can technically set any mode to any interface. What happens if you try to set a mode to tap (which should not be supported)?  Maybe you should try which mode can be set with some interface type and reflect it in the implementation.